### PR TITLE
Add document of linux_sbus driver

### PR DESCRIPTION
**Pull request by crossa**
Yesterday I've make a pull request of linux_sbus driver. https://github.com/PX4/Firmware/pull/7785
This driver require some  electronic component so I  write this document to help users to use this driver

### CODE REVIEW

**Comment by crossa** [comment]
@TSC21  here is the guide of linux_sbus driver

File: en/tutorials/linux_sbus.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# Sbus Driver for linux（drivers/linux_sbus）
+This driver allows the autopilot to fetch data from the futaba sbus receiver via serial port, up to 16 channels.
```
Comment (TSC21) [inline en/tutorials/linux_sbus.md]: Does this driver only supports Futaba? What about FrSky? This should be state here.

File: en/tutorials/linux_sbus.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+## Required components
+
+NPN transistor * 1  
+10K resistance * 1  
+1K resistor * 1  
+
+## Optional components
+Usbtty *1  
+
+## Installation
```
Comment (TSC21) [inline en/tutorials/linux_sbus.md]: `Installing`

File: en/tutorials/linux_sbus.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+Connect the components according to the following ways  
+
+S.bus singal->1K resistor->NPN transistor base  
+NPN transistor emmit -> GND  
+3.3v  VCC-> 10K resistor -> NPN transistor collection ->usbtty rxd  
+5.0v  VCC->S.bus vcc  
+GND ->S.bus gnd  
+
```
Comment (TSC21) [inline en/tutorials/linux_sbus.md]: Do you have an image of the actual circuit printed/soldered that you can add here?

File: en/tutorials/linux_sbus.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+## Circuit diagram
+![](http://www.playuav.com/uploads/article/20160310/56cf0f65bb1f7437c1618041a30dc308.png)
+
+## Launch the driver
+After the following command started, the autopilot would fetch the sbus data via /dev/ttyUSB0, up to 8 channels.
+```
+pxh> linux_sbus start -d /dev/ttyUSB0 -c 8
```
Comment (TSC21) [inline en/tutorials/linux_sbus.md]: This can also be automatically started on the board configs.

File: en/tutorials/linux_sbus.md [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+## Launch the driver
+After the following command started, the autopilot would fetch the sbus data via /dev/ttyUSB0, up to 8 channels.
+```
+pxh> linux_sbus start -d /dev/ttyUSB0 -c 8 
+```
+
+
+
+
```
Comment (TSC21) [inline en/tutorials/linux_sbus.md]: No need for all these extra nl

File: zh/tutorials/linux_sbus.md [hunk 6]
```diff
@@ -0,0 +1,35 @@
+
```
Comment (TSC21) [inline zh/tutorials/linux_sbus.md:1]: The changes must match the English version accordingly.

File: en/tutorials/linux_sbus.md [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+All platforms running linux system are supported.  
+
+## Required components
+
+NPN transistor * 1  
+10K resistance * 1  
+1K resistor * 1  
+
+## Optional components
+Usbtty *1
```
Comment (TSC21) [inline en/tutorials/linux_sbus.md]: I suppose you are making reference to a "USB to TTL Serial Cable/Converter". Needs to be more clear here.

File: en/tutorials/linux_sbus.md [hunk 8]
```diff
@@ -0,0 +1,35 @@
+# Sbus Driver for linux（drivers/linux_sbus）
+This driver allows the autopilot to fetch data from the futaba sbus receiver via serial port, up to 16 channels.  
+Onboard serial, usbtty, and other types of serial ports are supported.  
+All platforms running linux system are supported.  
+
+## Required components
+
+NPN transistor * 1
```
Comment (TSC21) [inline en/tutorials/linux_sbus.md]: What type of NPN transistor - can you recommend a part?

Comment (TSC21) [reply en/tutorials/linux_sbus.md]: And it looks better having quantity * part, instead of part * quantity.

File: en/tutorials/linux_sbus.md (hunk 3 above)
Comment (TSC21) [reply en/tutorials/linux_sbus.md]: Also you could share a PCB design so everyone can print their own

**Review by TSC21 — CHANGES_REQUESTED** [review]
@hamishwillee can you also give your opinion here please? Thank you

File: en/tutorials/linux_sbus.md (hunk 8 above)
Comment (crossa) [reply en/tutorials/linux_sbus.md]: Ok. 

I‘m editing this document  . Any type of NPN transistor can be used.

File: en/tutorials/linux_sbus.md (hunk 1 above)
Comment (crossa) [reply en/tutorials/linux_sbus.md]: Other receivers based on the sbus protocol are available, such as frsky, radioLink, and even sbus encoders.

File: en/tutorials/linux_sbus.md (hunk 7 above)
Comment (crossa) [reply en/tutorials/linux_sbus.md]: Yes.

File: en/tutorials/linux_sbus.md (hunk 3 above)
Comment (crossa) [reply en/tutorials/linux_sbus.md]: I did not make a PCB for this, just a handmade circuit.  I will draw a hand-painted circuit.
In fact this driver is also for "posx_rpi_no_shield"，It‘s inconvenient to buy pixfmini or navio in China. Since Last time I submitted the relevant code, many Chinese users began to  use the rpi to build the aircraft without shield. 

http://bbs.5imx.com/forum.php?mod=viewthread&tid=1252361&extra=page%3D2

File: en/tutorials/linux_sbus.md [hunk 9]
```diff
@@ -1,18 +1,21 @@
...
 All platforms running linux system are supported.  
 
 ## Required components
 
-NPN transistor * 1  
-10K resistance * 1  
-1K resistor * 1  
+1 * NPN transistor
+1 * 10K resistance
+1 * resistor * 1
```
Comment (TSC21) [inline en/tutorials/linux_sbus.md]: `1x 1K resistor`

File: en/tutorials/linux_sbus.md [hunk 10]
```diff
@@ -1,18 +1,21 @@
...
+You can also use other receivers based on the sbus protocol, such as frsky, radioLink, and even sbus encoders
+Onboard serial, USB to TTL serial cable/converter, and other types of serial ports are supported.  
 All platforms running linux system are supported.  
 
 ## Required components
 
-NPN transistor * 1  
-10K resistance * 1  
-1K resistor * 1  
+1 * NPN transistor
```
Comment (TSC21) [inline en/tutorials/linux_sbus.md]: looks better with `1x` instead of `1 * `

File: en/tutorials/linux_sbus.md (hunk 9 above)
Comment (crossa) [reply en/tutorials/linux_sbus.md]: Ok

File: en/tutorials/linux_sbus.md [hunk 11]
```diff
@@ -23,13 +26,3 @@ GND ->S.bus gnd
 
 ## Circuit diagram
 ![](http://www.playuav.com/uploads/article/20160310/56cf0f65bb1f7437c1618041a30dc308.png)
-
-## Launch the driver
```
Comment (TSC21) [inline en/tutorials/linux_sbus.md]: Why did you remove this? I just asked to show where to add the driver start on the config files.

Comment (crossa) [reply en/tutorials/linux_sbus.md]: Hold on a moment. I'm rewriting  it

File: en/tutorials/linux_sbus.md [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+1x  10K resistance  
+1x  1K  resistor  
+
+Any type of transistor can be used for this function. 
+
+## Optional components
+1x USB to TTL Serial Cable/Converter
+
+## Installing
+Connect the components according to the following ways
```
Comment (TSC21) [inline en/tutorials/linux_sbus.md]: `Connect the components according to the following` (remove `ways`)

File: en/tutorials/linux_sbus.md [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+Onboard serial, USB to TTL serial cable/converter, and other types of serial ports are supported.  
+All platforms running linux system are supported.  
+
+## Required components
+
+1x  NPN transistor  
+1x  10K resistance  
+1x  1K  resistor  
+
+Any type of transistor can be used for this function.
```
Comment (TSC21) [inline en/tutorials/linux_sbus.md]: replace "function" with "purpose"

**Comment by crossa** [comment]
@TSC21  Ok，it‘s done

**Comment by TSC21** [comment]
@hamishwillee can you leave your feedback please?

File: en/tutorials/linux_sbus.md [hunk 14]
```diff
@@ -0,0 +1,33 @@
+# Sbus Driver for linux（drivers/linux_sbus）
+This driver allows the autopilot to fetch data from the futaba sbus receiver via serial port, up to 16 channels.  
+You can also use other receivers based on the sbus protocol, such as frsky, radioLink, and even sbus encoders.  
+Onboard serial, USB to TTL serial cable/converter, and other types of serial ports are supported.  
+All platforms running linux system are supported.  
+
+## Required components
+
+1x  NPN transistor  
+1x  10K resistance
```
Comment (TSC21) [inline en/tutorials/linux_sbus.md]: resistor

File: en/tutorials/linux_sbus.md [hunk 15]
```diff
@@ -0,0 +1,33 @@
...
+
+S.bus singal->1K resistor->NPN transistor base  
+NPN transistor emmit -> GND  
+3.3v  VCC-> 10K resistor -> NPN transistor collection ->usbtty rxd  
+5.0v  VCC->S.bus vcc  
+GND ->S.bus gnd  
+
+## Circuit diagram
+![](http://www.playuav.com/uploads/article/20160310/56cf0f65bb1f7437c1618041a30dc308.png)
+![](https://github.com/crossa/raspx4-sbus-rc-in/blob/master/sbusToSerial_small.jpg?raw=true)
```
Comment (TSC21) [inline en/tutorials/linux_sbus.md]: We appreciate the diagram but what I was looking for was indeed an example of PCB or Perfboard with the components soldered (it serves as a graphical example). You can remove the drawing as the circuit diagram is self-explanatory.

File: zh/tutorials/linux_sbus.md [hunk 16]
```diff
@@ -25,7 +25,10 @@ GND ->S.bus GND
 
 ## 电路图
 ![](http://www.playuav.com/uploads/article/20160310/56cf0f65bb1f7437c1618041a30dc308.png)
-![](https://github.com/crossa/raspx4-sbus-rc-in/blob/master/sbusToSerial_small.jpg?raw=true)
+![](https://github.com/crossa/raspx4-sbus-rc-in/blob/master/sbusToSerial_small.jpg?raw=true)  
+
+## 安装实例
+![](https://raw.githubusercontent.com/crossa/raspx4-sbus-rc-in/master/example.jpeg)
```
Comment (TSC21) [inline zh/tutorials/linux_sbus.md]: Thanks for the example. Just a request: please identify on the image the different parts and connectors so people can identify it with the schematic you shared.

File: en/tutorials/linux_sbus.md [hunk 17]
```diff
@@ -0,0 +1,36 @@
...
+3.3v  VCC-> 10K resistor -> NPN transistor collection ->usbtty rxd  
+5.0v  VCC->S.bus vcc  
+GND ->S.bus gnd  
+
+## Circuit diagram
+![](http://www.playuav.com/uploads/article/20160310/56cf0f65bb1f7437c1618041a30dc308.png)
+![](https://github.com/crossa/raspx4-sbus-rc-in/blob/master/sbusToSerial_small.jpg?raw=true)  
+
+## Example
+![](https://raw.githubusercontent.com/crossa/raspx4-sbus-rc-in/master/example.jpeg)
```
Comment (TSC21) [inline en/tutorials/linux_sbus.md]: Thanks for the example. Just a request: please identify on the image the different parts and connectors so people can identify it with the schematic you shared.

Comment (crossa) [reply en/tutorials/linux_sbus.md]: Ok, I have updated the picture. I think the user can make the device according to the circuit diagram and example.

File: en/tutorials/linux_sbus.md [hunk 18]
```diff
@@ -29,7 +29,7 @@ GND ->S.bus gnd
 ![](https://github.com/crossa/raspx4-sbus-rc-in/blob/master/sbusToSerial_small.jpg?raw=true)  
 
 ## Example
-![](https://raw.githubusercontent.com/crossa/raspx4-sbus-rc-in/master/example.jpeg)
+![](https://raw.githubusercontent.com/crossa/raspx4-sbus-rc-in/master/example.png)
```
Comment (TSC21) [inline en/tutorials/linux_sbus.md]: Although not perfect, seems acceptable to me. Thanks for the contribution @crossa

**Review by TSC21 — APPROVED** [review]

**Comment by TSC21** [comment]
@hamishwillee your call now

File: en/tutorials/linux_sbus.md [hunk 19]
```diff
@@ -0,0 +1,36 @@
...
+
+## Circuit diagram
+![](http://www.playuav.com/uploads/article/20160310/56cf0f65bb1f7437c1618041a30dc308.png)
+![](https://github.com/crossa/raspx4-sbus-rc-in/blob/master/sbusToSerial_small.jpg?raw=true)  
+
+## Example
+![](https://raw.githubusercontent.com/crossa/raspx4-sbus-rc-in/master/example.png)
+
+## Launch it  
+Add `linux_sbus start -d /dev/ttyUSB0 -c 8` to the configuration file and it will start automatically
```
Comment (TSC21) [inline en/tutorials/linux_sbus.md]: `to the configuration file and it will start automatically the driver on /dev/ttyUSB0 listening to 8 channels.`

**Comment by TSC21** [comment]
@crossa also I don't see any references here about using the `PCA9685` and how all should be wired to it.

File: en/tutorials/linux_sbus.md [hunk 20]
```diff
@@ -0,0 +1,36 @@
...
+
+Any type of transistor can be used for this purpose. 
+
+## Optional components
+1x USB to TTL Serial Cable/Converter
+
+## Installing
+Connect the components according to the following  
+
+S.bus singal->1K resistor->NPN transistor base
```
Comment (TSC21) [inline en/tutorials/linux_sbus.md:25]: replace with: `S.bus signal -> 1K resistor -> NPN transistor base`

File: en/tutorials/linux_sbus.md [hunk 21]
```diff
@@ -0,0 +1,36 @@
...
+
+## Optional components
+1x USB to TTL Serial Cable/Converter
+
+## Installing
+Connect the components according to the following  
+
+S.bus singal->1K resistor->NPN transistor base  
+NPN transistor emmit -> GND  
+3.3v  VCC-> 10K resistor -> NPN transistor collection ->usbtty rxd
```
Comment (TSC21) [inline en/tutorials/linux_sbus.md]: `3.3VCC -> 10K resistor -> NPN transistor collection -> USB-to-TTY rxd`

File: en/tutorials/linux_sbus.md [hunk 22]
```diff
@@ -0,0 +1,36 @@
...
+## Optional components
+1x USB to TTL Serial Cable/Converter
+
+## Installing
+Connect the components according to the following  
+
+S.bus singal->1K resistor->NPN transistor base  
+NPN transistor emmit -> GND  
+3.3v  VCC-> 10K resistor -> NPN transistor collection ->usbtty rxd  
+5.0v  VCC->S.bus vcc
```
Comment (TSC21) [inline en/tutorials/linux_sbus.md]: `5VCC -> S.bus VCC`

File: en/tutorials/linux_sbus.md [hunk 23]
```diff
@@ -0,0 +1,36 @@
...
+1x USB to TTL Serial Cable/Converter
+
+## Installing
+Connect the components according to the following  
+
+S.bus singal->1K resistor->NPN transistor base  
+NPN transistor emmit -> GND  
+3.3v  VCC-> 10K resistor -> NPN transistor collection ->usbtty rxd  
+5.0v  VCC->S.bus vcc  
+GND ->S.bus gnd
```
Comment (TSC21) [inline en/tutorials/linux_sbus.md]: `GND -> S.bus GND`

File: en/tutorials/linux_sbus.md [hunk 24]
```diff
@@ -0,0 +1,36 @@
+# Sbus Driver for linux（drivers/linux_sbus）
+This driver allows the autopilot to fetch data from the futaba sbus receiver via serial port, up to 16 channels.  
+You can also use other receivers based on the sbus protocol, such as frsky, radioLink, and even sbus encoders.
```
Comment (TSC21) [inline en/tutorials/linux_sbus.md]: Use capital letters along the guide please: `FrSky`, `RadioLink`, `SBUS`, `Futaba`.

File: en/tutorials/linux_sbus.md [hunk 25]
```diff
@@ -0,0 +1,36 @@
...
+Connect the components according to the following  
+
+S.bus singal->1K resistor->NPN transistor base  
+NPN transistor emmit -> GND  
+3.3v  VCC-> 10K resistor -> NPN transistor collection ->usbtty rxd  
+5.0v  VCC->S.bus vcc  
+GND ->S.bus gnd  
+
+## Circuit diagram
+![](http://www.playuav.com/uploads/article/20160310/56cf0f65bb1f7437c1618041a30dc308.png)
```
Comment (hamishwillee) [inline en/tutorials/linux_sbus.md]: 1. Why two circuit diagrams? If these are the same, then just one required - best that it be drawn using a tool and labelled correctly 
2. Ideally it would also clearly label the autopilot and receiver ends.

File: en/tutorials/linux_sbus.md [hunk 26]
```diff
@@ -0,0 +1,36 @@
+# Sbus Driver for linux（drivers/linux_sbus）
```
Comment (hamishwillee) [inline en/tutorials/linux_sbus.md]: Title should be "SBUS Driver for Linux"

I'm guessing that the drivers/linux_sbus）info is where software for driver is stored? If so, that should be stated in the body of the doc.

File: en/tutorials/linux_sbus.md [hunk 27]
```diff
@@ -0,0 +1,36 @@
...
+
+## Circuit diagram
+![](http://www.playuav.com/uploads/article/20160310/56cf0f65bb1f7437c1618041a30dc308.png)
+![](https://github.com/crossa/raspx4-sbus-rc-in/blob/master/sbusToSerial_small.jpg?raw=true)  
+
+## Example
+![](https://raw.githubusercontent.com/crossa/raspx4-sbus-rc-in/master/example.png)
+
+## Launch it  
+Add `linux_sbus start -d /dev/ttyUSB0 -c 8` to the configuration file and it will start automatically the driver on /dev/ttyUSB0 listening to 8 channels.
```
Comment (hamishwillee) [inline en/tutorials/linux_sbus.md]: What configuration file? Can we name it so that a normal Linux user would be able to find it, or a beginner who has never used PX4 on Linux?

File: en/tutorials/linux_sbus.md [hunk 28]
```diff
@@ -0,0 +1,36 @@
...
+3.3v  VCC-> 10K resistor -> NPN transistor collection ->usbtty rxd  
+5.0v  VCC->S.bus vcc  
+GND ->S.bus gnd  
+
+## Circuit diagram
+![](http://www.playuav.com/uploads/article/20160310/56cf0f65bb1f7437c1618041a30dc308.png)
+![](https://github.com/crossa/raspx4-sbus-rc-in/blob/master/sbusToSerial_small.jpg?raw=true)  
+
+## Example
+![](https://raw.githubusercontent.com/crossa/raspx4-sbus-rc-in/master/example.png)
```
Comment (hamishwillee) [inline en/tutorials/linux_sbus.md]: If a used image might move (?), you should pull it into the docs tree rather than linking like this.  You'd give it a sensible name so it would be easy to find and logically located.

File: en/tutorials/linux_sbus.md [hunk 29]
```diff
@@ -0,0 +1,36 @@
...
+## Required components
+
+1x  NPN transistor  
+1x  10K resistor  
+1x  1K  resistor  
+
+Any type of transistor can be used for this purpose. 
+
+## Optional components
+1x USB to TTL Serial Cable/Converter
```
Comment (hamishwillee) [inline en/tutorials/linux_sbus.md]: Link to an example we know would work?

File: en/tutorials/linux_sbus.md [hunk 30]
```diff
@@ -0,0 +1,36 @@
...
+GND ->S.bus gnd  
+
+## Circuit diagram
+![](http://www.playuav.com/uploads/article/20160310/56cf0f65bb1f7437c1618041a30dc308.png)
+![](https://github.com/crossa/raspx4-sbus-rc-in/blob/master/sbusToSerial_small.jpg?raw=true)  
+
+## Example
+![](https://raw.githubusercontent.com/crossa/raspx4-sbus-rc-in/master/example.png)
+
+## Launch it
```
Comment (hamishwillee) [inline en/tutorials/linux_sbus.md:39]: Will it always be connected to /dev/ttyUSB0 - can this be connected to any serial pins?

**Comment by hamishwillee** [comment]
I think of a driver as some software that lives between a hardware abstraction layer and some real hardware. It is hard to tell looking at this article that it is about that software - the whole doc is about some circuit, the purpose of which is never explained.

What I THINK this is saying is that:
* There is a driver that can interface with any SBUS receiver - correct?
* There is a circuit you need to convert whatever is coming out of the receiver so it can connect to the linux board - correct?
  * Why do you need this? I'm GUESSING  the linux board has a lower voltage than expected by the receiver, and this just steps up the signal. 
  * Are there receivers where you would not need this?
* The "USB to TTL Serial Cable/Converter" is optional. I am guessing this does the same thing as the circuit but is ready made - correct? If so, would be good to provide a source.

**Comment by hamishwillee** [comment]
PS This is new right? You will need to update the SUMMARY.md to get it to build in gitbook.

File: en/tutorials/linux_sbus.md (hunk 30 above)
Comment (crossa) [reply en/tutorials/linux_sbus.md:39]: Yes. You can use another  serial device to replace it such as /dev/ttyS0.
Initially I was using the onboard serial port to connect it.

File: en/tutorials/linux_sbus.md (hunk 29 above)
Comment (crossa) [reply en/tutorials/linux_sbus.md]: Yes,This example is installed on my copter

File: en/tutorials/linux_sbus.md (hunk 26 above)
Comment (crossa) [reply en/tutorials/linux_sbus.md]: OK

File: en/tutorials/linux_sbus.md (hunk 25 above)
Comment (crossa) [reply en/tutorials/linux_sbus.md]: In fact I do not have the tools to draw circuit diagrams in my daily work system and I'm not an electronic engineer. I'm not familiar with this kind of software.I made these devices only refer to these simple schematics.This driver will benefit all users who want to use the Linux system as a flight system. If I can, I hope to get some help.

File: en/tutorials/linux_sbus.md (hunk 26 above)
Comment (crossa) [reply en/tutorials/linux_sbus.md]: Yes

Comment (crossa) [reply en/tutorials/linux_sbus.md]: I've changed it

File: en/tutorials/linux_sbus.md (hunk 28 above)
Comment (crossa) [reply en/tutorials/linux_sbus.md]: I've removed it

**Comment by crossa** [comment]
@hamishwillee Since I am not an electronic engineer, it is difficult for me to explain the principle of this device in English. I try to explain it.
1. The S.bus receiver is a reverse serial port and s.bus signal is the reverse of the serial signal
2. This circuit is a director circuit for converting the sbus receiver signal into an ordinary serial signal.
3. This circuit is a required otherwise the s.bus signal can not be resolved.
4. USB-to-tty is used to receive the back of the sbus signal, you can also use the onboard serial port to instead

Please tell me how the above content is correctly expressed in written language, I can write these into the document.

thank you

File: en/tutorials/linux_sbus.md (hunk 27 above)
Comment (crossa) [reply en/tutorials/linux_sbus.md]: Let me think about it. This is not  easy ti explain.  Some users use navtive way to compile px4, after compiling,  they do not make "make upload" operation. In this case I can not describe the location of the configuration file. The original configuration file is located in  Firmware/posix-configs and there are 4 files in the directory. Can you give me some advice?

File: en/tutorials/linux_sbus.md [hunk 31]
```diff
@@ -0,0 +1,36 @@
...
+## Required components
+
+1x  NPN transistor  
+1x  10K resistor  
+1x  1K  resistor  
+
+Any type of transistor can be used for this purpose. 
+
+## Optional components
+1x USB to TTL Serial Cable/Converter
```
Comment (hamishwillee) [inline en/tutorials/linux_sbus.md]: 1. I mean "Link to a real off the shelf component that someone can buy" - unless "USB to TTL Serial Cable/Converter" is a completely unambiguous term that someone can find in any electronics store. 
2. Why is it optional? What if you don't have it? Does that change the circuit, and what does it change otherwise?

File: en/tutorials/linux_sbus.md [hunk 32]
```diff
@@ -0,0 +1,36 @@
+# SBus Driver for Linux
+This driver allows the autopilot to fetch data from the futaba sbus receiver via serial port, up to 16 channels.  
+You can also use other receivers based on the sbus protocol, such as FrSky, radioLink, and even sbus encoders.  
+Onboard serial, USB to TTL serial cable/converter, and other types of serial ports are supported.  
+All platforms running linux system are supported.  
+`This driver is located in  Firm/src/drivers/linux_sbus`
+
+## Required components
+
+1x  NPN transistor
```
Comment (hamishwillee) [inline en/tutorials/linux_sbus.md]: Is this unambiguous? If I go to a store will I have a choice of NPN transistors? If it is ambiguous, are there any particular  requirements on the transistor?

File: en/tutorials/linux_sbus.md (hunk 27 above)
Comment (hamishwillee) [reply en/tutorials/linux_sbus.md]: Yes, refer to the original file by location and name. Presumably they will start from that wherever it gets moved via their build process.

File: en/tutorials/linux_sbus.md (hunk 31 above)
Comment (crossa) [reply en/tutorials/linux_sbus.md]: In taobao and In china  , the "usb to tty" is not ambiguous, almost the same product.
Ok, I recommend  PL2302

**Comment by hamishwillee** [comment]
HI @crossa 

OK, from https://oscarliang.com/pwm-ppm-sbus-dsm2-dsmx-sumd-difference/
"SBUS is an inverted UART communication signal. Many flight controllers can read UART input, but cannot accept inverted one (such as the Naze32 Rev5) and an inverted is required. However F3 FC’s and some FC such as Pixhawks has built-in dedicated signal inverter for this purpose."

So this circuit is a signal inverter. Presumably if the linux system you had could read inverted UART input you wouldn't need it. 

> USB-to-tty is used to receive the back of the sbus signal, you can also use the onboard serial port to instead

I still don't understand. So the circuit converts the SBUS signal from the receiver TO the linux board (RaPi?) while the USB-to-TTY converts back from the board to the receiver? It isn't really clear in the diagram. 

Might be clearer if the circuit diagram included the Receiver and Flight controller and it showed how the UBS-TTY cable fits in.  Either way, you need to explain in the doc why it is optional/what it is for, and what configuration is required if it is present or not present.

ADDED:
Assume you are someone like me and you come to this document. 
- Can you get all the electronics you need
- Can you SEE how they are wired up or work it out from the diagram

**Comment by crossa** [comment]
@hamishwillee  Sorry it's a mistake. I means USB-to-tty is used to receive the inverted sbus signal.
There is only one serial port on rpi and we often use this serial port to login into the system, so we need usb -to-tty convert as another serial port to communicate with the device such  as GPS, reciver and any other serial device. USB-to-tty  converter do not send any data to the reciver.

Ok  thank you I'll change the document later

**Comment by crossa** [comment]
By the way,these components are very common, easy to buy :-)

File: en/tutorials/linux_sbus.md (hunk 20 above)
Comment (crossa) [reply en/tutorials/linux_sbus.md:25]: done

File: en/tutorials/linux_sbus.md (hunk 27 above)
Comment (crossa) [reply en/tutorials/linux_sbus.md]: If they follow the documentation, these configuration files will located in /home/pi/ 
Add the command to the file which they used.

**Comment by crossa** [comment]
@hamishwillee it's down ,pleas check it again

File: en/SUMMARY.md [hunk 33]
```diff
@@ -44,6 +44,7 @@
   * [Sensor Thermal Compensation](tutorials/sensor_thermal_calibration.md)
   * [Land Detector](tutorials/land_detector.md)
   * [Motion Capture \(VICON, Optitrack\)](tutorials/motion-capture-vicon-optitrack.md)
+  * [S.bus driver for Linux](tutorials/linux_sbus.md)
```
Comment (hamishwillee) [inline en/SUMMARY.md]: SBus Driver for Linux

File: en/tutorials/linux_sbus.md [hunk 34]
```diff
@@ -0,0 +1,43 @@
+# SBus Driver for Linux
```
Comment (hamishwillee) [inline en/tutorials/linux_sbus.md:1]: As previously commented (not by me) proper nouns (names) are capitalised. E.g. Futaba, not futaba, Linux. Similarly, protocols have proper spellings (I *think* SBus, not sbus)

File: en/tutorials/linux_sbus.md [hunk 35]
```diff
@@ -0,0 +1,43 @@
...
+This driver allows the autopilot to fetch data from the futaba sbus receiver via serial port, up to 16 channels.  
+You can also use other receivers based on the sbus protocol, such as FrSky, radioLink, and even sbus encoders.  
+Onboard serial, USB to TTL serial cable/converter, and other types of serial ports are supported.  
+All platforms running linux system are supported.  
+`This driver is located in  Firm/src/drivers/linux_sbus`
```
Comment (hamishwillee) [inline en/tutorials/linux_sbus.md]: Change
```
This driver is located in  Firm/src/drivers/linux_sbus`
```
To remove backticks and make like below
```
This driver is located in: **Firmware/src/drivers/linux_sbus**
```

File: en/tutorials/linux_sbus.md [hunk 36]
```diff
@@ -0,0 +1,43 @@
...
+
+## Required components
+
+1x  NPN transistor  (I recommend NPN S9014 TO92)  
+1x  10K resistor  
+1x  1K  resistor  
+
+`Because of the load current is small ,any type of transistor can be used for this purpose. `
```
Comment (hamishwillee) [inline en/tutorials/linux_sbus.md]: For this sort of thing, don't use backtick. You can use a note like this
```
> **Note** Because of the load...
```
That renders as a nice box

**Comment by hamishwillee** [comment]
I have added a few comments. I think perhaps on Monday I should merge this, and make modifications I see as necessary (for you to comment on).

**Comment by hamishwillee** [comment]
It would still be good if you could fully label the circuit diagram. As an amateur I don't know if  RC_SBUS and the other values are on the receiver or the board. If you just add RaPi and Reciever to the outputs that would make thing clearer

**Comment by crossa** [comment]
Ok. I've modified the document as necessary. I'll learn to use fritzing to draw the circuit diagram but it's too late to do this work this moment.  I'll update this document in the furture.

**Comment by hamishwillee** [comment]
I have merged this, but am going to do further corrections. Will add PR link for that shortly. 

@TSC21 Were you happy adequate changes were made to https://github.com/PX4/Devguide/pull/257#issuecomment-322856727

@crossa 
>  I'll learn to use fritzing to draw the circuit diagram but it's too late to do this work this moment. I'll update this document in the furture.

Fair enough. When?

**Comment by crossa** [comment]
@hamishwillee  about two weeks later.

**Comment by hamishwillee** [comment]
OK, thank you.

**Comment by hamishwillee** [comment]
@crossa What linux platform did you run on the RaPi/test the driver with. I understand that it *should* work on all Linux platforms, but good to state what it has been tested on.

**Comment by hamishwillee** [comment]
@crossa Is it USB to TTL or USB to TTY - both are referred to in the text.

**Comment by crossa** [comment]
Ok. In fact  this driver can be tested on a desktop computer with linux system such as  Debian,Ubuntu and users can also test it on raspberry pi with raspbian or any other Linux system.
USB-to-TTL is one of the actual product names of the USB-to-TTY converter.

http://www.ebay.com/itm/New-PL2303HX-USB-to-TTL-RS232-COM-UART-Module-Serial-Cable-Adapter-for-Arduino-/111593270026?epid=744201922&hash=item19fb7a2b0a:g:SFgAAOSwTM5Yz1GH
